### PR TITLE
Added interpolation for TsDataframe

### PR DESCRIPTION
**Pull request by vigji**
Currently interpolation was implemented only for `Tsd`, but it can be very useful for `TsDataframe` as well and I do not see why not to extend it, so I did that. If you think this might be useful to merge & not something that is already happening somewhere else, let me know and I can write a test function.

### CODE REVIEW

**Comment by gviejo** [comment]
Yes it would be nice but I would move the function to `_AbstractTsd` and make it generic for Tsd, TsdTensor and TsdFrame then.

**Comment by vigji** [comment]
Should be good to go now! Copied tests from existing material for Tsd.

File: pynapple/core/time_series.py [hunk 1]
```diff
@@ -927,6 +927,61 @@ def smooth(self, std, size):
...
+                        tmp.index.values,
+                        tmp.values,
+                        left=left,
+                        right=right,
+                    )
+                else:
+                    interpolated_values = np.apply_along_axis(
+                        lambda row: np.interp(t.index.values, tmp.index.values, row),
+                        0,
+                        tmp.values,
```
Comment (gviejo) [inline pynapple/core/time_series.py:977]: The left right arguments are missing here.

**Review by gviejo — CHANGES_REQUESTED** [review]

File: pynapple/core/time_series.py (hunk 1 above)
Comment (vigji) [reply pynapple/core/time_series.py:977]: Sorry what do you mean here?

Comment (vigji) [reply pynapple/core/time_series.py:977]: Oh got it sorry

File: pynapple/core/time_series.py [hunk 2]
```diff
@@ -927,6 +927,61 @@ def smooth(self, std, size):
...
+        new_t = ts.restrict(ep).index
+
+        new_shape = (
+            len(new_t) if self.values.ndim == 1 else (len(new_t),) + self.shape[1:]
+        )
+        new_d = np.full(new_shape, np.nan)
+
+        start = 0
+        for i in range(len(ep)):
+            t = ts.restrict(ep.loc[[i]])
```
[the remark this answers is not in the document]
Comment (vigji) [reply pynapple/core/time_series.py]: Done

File: tests/test_time_series.py [hunk 3]
```diff
@@ -7,11 +7,14 @@
 
 """Tests of time series for `pynapple` package."""
 
+from turtle import st
```
Comment (gviejo) [inline tests/test_time_series.py:10]: Hi is it important this?

Comment (vigji) [reply tests/test_time_series.py:10]: oh sorry it is completely random?

**Review by gviejo — APPROVED** [review]

**Comment by gviejo** [comment]
Hey Luigi, I can't merge your branch. Could you update your fork to the latest commits?

**Comment by gviejo** [comment]
Thanks

**Pull request merged**